### ISSUE TITLE
[no-unused-vars] TypeError: Cannot read property 'type' of undefined

### ISSUE THREAD

**Issue by CharlyPoppins** [typescript-eslint/typescript-eslint#1246]
**What code were you trying to parse?**

```ts
// Somewhere in the file I have got
import User from '../../../model/user'
this.myFunction(if).then(
    (object: User) => {
            //
    },
    (error: any)  => {
            //
     },
)
// error is trigger on the line with the import
```

**What did you expect to happen?**
It gives me an handled error return :)

**What actually happened?**
```bash
TypeError: Cannot read property 'type' of undefined
Occurred while linting **myFile.ts:1
    at isForInRef (node_modules/eslint/lib/rules/no-unused-vars.js:423:24)
    at node_modules/eslint/lib/rules/no-unused-vars.js:460:21
    at Array.some (<anonymous>)
    at isUsedVariable (node_modules/eslint/lib/rules/no-unused-vars.js:459:40)
    at collectUnusedVariables (node_modules/eslint/lib/rules/no-unused-vars.js:570:26)
    at collectUnusedVariables (node_modules/eslint/lib/rules/no-unused-vars.js:577:17)
    at Program:exit (/node_modules/eslint/lib/rules/no-unused-vars.js:589:36)
    at node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (node_modules/eslint/lib/linter/safe-emitter.js:45:38)
```
**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.8.0` |
| `@typescript-eslint/eslint-plugin` | `2.8.0` |
| `TypeScript`                | `3.7.2` |
| `ESLint`                    | `v6.6.0` |
| `node`                      | `v12.4.0` |
| `npm`                       | `6.13.0` |

**Comment by bradzacher** [comment]
could you please confirm that you are using `@typescript-eslint/no-unused-vars`?

Also, could you please provide an isolated repro case?
The snippet you provided works fine in master.

**Comment by CharlyPoppins** [comment]
I wasn't but I do now (according to https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md)

For now I can't provide a real example, need to ask my boss and our customer.

Also I can't reproduce the initial error... The error is now triggered on the import. Does not matter if i'm using the imported class or not.

**Comment by bradzacher** [comment]
In that case, is a KP. https://github.com/typescript-eslint/typescript-eslint/issues/363
